pkg/controller/core.oam.dev/v1alpha2/application/dispatch/dispatch.go

```diff
@@ -69,6 +69,7 @@ type AppManifestsDispatcher struct {
 func (a *AppManifestsDispatcher) EnableUpgradeAndGC(rt *v1beta1.ResourceTracker) *AppManifestsDispatcher {
 	if rt != nil {
 		a.previousRT = rt.DeepCopy()
+		a.skipGC = false
```

Comment:
do we have bug here before?

Reply:
yes.  If we enable skip GC at first and enable GC later. Enable GC will not work for `a.skipGC = true`.

Reply:
But before we didn't use such a case.